### PR TITLE
Update Contract_signing test suite

### PR DESCRIPTION
**Pull request by OrysiaDrabych**
This change is [Reviewable](https://reviewable.io/reviews/openprocurement/robot_tests/457)

### CODE REVIEW

File: op_robot_tests/tests_files/brokers/openprocurement_client.robot [hunk 1]
```diff
@@ -1066,3 +1053,78 @@ Library  openprocurement_client_helper.py
...
+
+##############################################################################
+#             CONTRACT SIGNING
+##############################################################################
+
+Редагувати угоду
+  [Arguments]  ${username}  ${tender_uaid}  ${contract_index}  ${fieldname}  ${fieldvalue}
+  ${tender}=  openprocurement_client.Пошук тендера по ідентифікатору  ${username}  ${tender_uaid}
+  ${contract}=  Create Dictionary  data=${tender.data.contracts[${contract_index}]}
+  Remove From Dictionary  ${USERS.users['${viewer}'].tender_data.data.contracts[${contract_index}].value}  amount
```
Comment (kosaniak) [inline op_robot_tests/tests_files/brokers/openprocurement_client.robot]: чистимо тут кеш користувача, щоб редагувати поле. Але 
1. Якщо я захочу редагувати якесь інше поле, не amount, то відповідно маю дописати ще рядок з Remove From Dictionary.  Ще одне поле редагую = ще мушу додати Remove From Dictionary. Можна це зробити універсальніше, для будь-якого поля. 
2. Як у себе має почистити кеш майданчик, якщо у нас він чиститься у клієнті?

File: op_robot_tests/tests_files/contract_signing.robot [hunk 2]
```diff
@@ -32,28 +31,142 @@ Suite Teardown  Test Suite Teardown
...
+  Run As  ${tender_owner}  Вказати дату початку дії угоди  ${TENDER['TENDER_UAID']}  -1
+
+
+Відображення дати початку дії угоди
+  [Tags]  ${USERS.users['${viewer}'].broker}: Відображення основних даних угоди
+  ...      viewer
+  ...      ${USERS.users['${viewer}'].broker}
+  ...      contract_sign
+  [Setup]  Дочекатись синхронізації з майданчиком  ${viewer}
+  Звірити відображення поля contracts[-1].period.startDate тендера із ${USERS.users['${tender_owner}'].contract_startDate} для користувача ${viewer}
```
Comment (kosaniak) [inline op_robot_tests/tests_files/contract_signing.robot]: В майданчиків це не працюватиме, бо contract_startDate записується в ${USERS.users['${tender_owner}']} в клієнті і вони цього не побачуть

File: op_robot_tests/tests_files/contract_signing.robot [hunk 3]
```diff
@@ -32,28 +31,142 @@ Suite Teardown  Test Suite Teardown
...
+  Run As  ${tender_owner}  Вказати дату завершення дії угоди  ${TENDER['TENDER_UAID']}  -1
+
+
+Відображення дати завершення дії угоди
+  [Tags]  ${USERS.users['${viewer}'].broker}: Відображення основних даних угоди
+  ...      viewer
+  ...      ${USERS.users['${viewer}'].broker}
+  ...      contract_sign
+  [Setup]  Дочекатись синхронізації з майданчиком  ${viewer}
+  Звірити відображення поля contracts[-1].period.endDate тендера із ${USERS.users['${tender_owner}'].contract_endDate} для користувача ${viewer}
```
Comment (kosaniak) [inline op_robot_tests/tests_files/contract_signing.robot]: Аналогічно і тут

File: op_robot_tests/tests_files/service_keywords.py [hunk 4]
```diff
@@ -410,6 +412,10 @@ def get_id_from_object(obj):
     return obj_id.group(1)
 
 
+def get_id_from_doc_name(name):
```
Comment (kosaniak) [inline op_robot_tests/tests_files/service_keywords.py]: а ця функція, яка нижче не працюватиме для вичитування id з документа?

File: op_robot_tests/tests_files/contract_signing.robot [hunk 5]
```diff
@@ -1,6 +1,5 @@
 *** Settings ***
-Resource        keywords.robot
-Resource        resource.robot
```
Comment (OrysiaDrabych) [inline op_robot_tests/tests_files/contract_signing.robot]: Щоб не дублювати. Resource.robot вже імпортується в base_keywords.robot

**Comment by OrysiaDrabych** [comment]
Review status: 0 of 6 files reviewed at latest revision, 11 unresolved discussions.

---

*[op_robot_tests/tests_files/contract_signing.robot, line 12 at r6](https://reviewable.io:443/reviews/openprocurement/robot_tests/457#-Kbz_SUxpiMIFC3FnuPM-r6-12:-KhfzRdHBmRnwlSa0d5C:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/c724750af6ac760c4ae6c61f96c007d1ad999e5e/op_robot_tests/tests_files/contract_signing.robot#L12)):*
<details><summary><i>Previously, kosaniak wrote…</i></summary><blockquote>

вирівняти треба, між [Tags] i ${USERS.users['${viewer}'].broker} три пробіла має бути
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/contract_signing.robot, line 25 at r6](https://reviewable.io:443/reviews/openprocurement/robot_tests/457#-Kbz_SUxpiMIFC3FnuPM-r6-25:-KhfzQCcXYcUVA7y6OXc:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/c724750af6ac760c4ae6c61f96c007d1ad999e5e/op_robot_tests/tests_files/contract_signing.robot#L25)):*
<details><summary><i>Previously, kosaniak wrote…</i></summary><blockquote>

тут
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/contract_signing.robot, line 34 at r6](https://reviewable.io:443/reviews/openprocurement/robot_tests/457#-Kbz_SUxpiMIFC3FnuPM-r6-34:-KhfzPbG_3SlXDVWpBD4:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/c724750af6ac760c4ae6c61f96c007d1ad999e5e/op_robot_tests/tests_files/contract_signing.robot#L34)):*
<details><summary><i>Previously, kosaniak wrote…</i></summary><blockquote>

і тут
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/contract_signing.robot, line 43 at r6](https://reviewable.io:443/reviews/openprocurement/robot_tests/457#-Kbz_SUxpiMIFC3FnuPM-r6-43:-KhfzP6FmMSYwSG6XKo1:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/c724750af6ac760c4ae6c61f96c007d1ad999e5e/op_robot_tests/tests_files/contract_signing.robot#L43)):*
<details><summary><i>Previously, kosaniak wrote…</i></summary><blockquote>

тут
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/contract_signing.robot, line 55 at r6](https://reviewable.io:443/reviews/openprocurement/robot_tests/457#-Kbz_SUxpiMIFC3FnuPM-r6-55:-KhfzORS3X4OR-IsiyW0:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/c724750af6ac760c4ae6c61f96c007d1ad999e5e/op_robot_tests/tests_files/contract_signing.robot#L55)):*
<details><summary><i>Previously, kosaniak wrote…</i></summary><blockquote>

всюди)
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/contract_signing.robot, line 79 at r6](https://reviewable.io:443/reviews/openprocurement/robot_tests/457#-Kbz_SUxpiMIFC3FnuPM-r6-79:-KhfzNnro5euRy2V6MYs:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/c724750af6ac760c4ae6c61f96c007d1ad999e5e/op_robot_tests/tests_files/contract_signing.robot#L79)):*
<details><summary><i>Previously, kosaniak wrote…</i></summary><blockquote>

 і тут
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/contract_signing.robot, line 98 at r6](https://reviewable.io:443/reviews/openprocurement/robot_tests/457#-Kbz_SUxpiMIFC3FnuPM-r6-98:-KhfzNGaH67v8vC17y9z:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/c724750af6ac760c4ae6c61f96c007d1ad999e5e/op_robot_tests/tests_files/contract_signing.robot#L98)):*
<details><summary><i>Previously, kosaniak wrote…</i></summary><blockquote>

В майданчиків це не працюватиме, бо contract_startDate записується в ${USERS.users['${tender_owner}']} в клієнті і вони цього не побачуть
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/contract_signing.robot, line 117 at r6](https://reviewable.io:443/reviews/openprocurement/robot_tests/457#-Kbz_SUxpiMIFC3FnuPM-r6-117:-KhfzLhUE3nXEhdQA95N:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/c724750af6ac760c4ae6c61f96c007d1ad999e5e/op_robot_tests/tests_files/contract_signing.robot#L117)):*
<details><summary><i>Previously, kosaniak wrote…</i></summary><blockquote>

Аналогічно і тут
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/initial_data.py, line 31 at r6](https://reviewable.io:443/reviews/openprocurement/robot_tests/457#-Kbz_SUxpiMIFC3FnuPN-r6-31:-KhfyyXvHMbt-9uxUMGT:b2o0ncd) ([raw file](https://github.com/openprocurement/robot_tests/blob/c724750af6ac760c4ae6c61f96c007d1ad999e5e/op_robot_tests/tests_files/initial_data.py#L31)):*
<details><summary><i>Previously, kosaniak wrote…</i></summary><blockquote>

Немає обмежень на верхню межу вартості угоди?
</blockquote></details>

Додала верхню межу

---

*[op_robot_tests/tests_files/service_keywords.py, line 415 at r6](https://reviewable.io:443/reviews/openprocurement/robot_tests/457#-Kbz_SUxpiMIFC3FnuPO-r6-415:-KhfymqoLNCAFwtH04n6:bbzym3f) ([raw file](https://github.com/openprocurement/robot_tests/blob/c724750af6ac760c4ae6c61f96c007d1ad999e5e/op_robot_tests/tests_files/service_keywords.py#L415)):*
<details><summary><i>Previously, kosaniak wrote…</i></summary><blockquote>

а ця функція, яка нижче не працюватиме для вичитування id з документа?
</blockquote></details>

Ти права, ця нова функція зайва

---

*[op_robot_tests/tests_files/brokers/openprocurement_client.robot, line 1065 at r6](https://reviewable.io:443/reviews/openprocurement/robot_tests/457#-Kbz_SUxpiMIFC3FnuPL-r6-1065:-Khfz36rY4X8I8wwXvc_:bwx05nl) ([raw file](https://github.com/openprocurement/robot_tests/blob/c724750af6ac760c4ae6c61f96c007d1ad999e5e/op_robot_tests/tests_files/brokers/openprocurement_client.robot#L1065)):*
<details><summary><i>Previously, kosaniak wrote…</i></summary><blockquote>

чистимо тут кеш користувача, щоб редагувати поле. Але 
1. Якщо я захочу редагувати якесь інше поле, не amount, то відповідно маю дописати ще рядок з Remove From Dictionary.  Ще одне поле редагую = ще мушу додати Remove From Dictionary. Можна це зробити універсальніше, для будь-якого поля. 
2. Як у себе має почистити кеш майданчик, якщо у нас він чиститься у клієнті?
</blockquote></details>

винесла на рівень вище, щоб майданчик зміг почистити кеш

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests/457)*

File: op_robot_tests/tests_files/brokers/openprocurement_client.robot [hunk 6]
```diff
@@ -1069,3 +1056,71 @@ Library  openprocurement_client_helper.py
...
+  Set To Dictionary  ${contract.data}  dateSigned=${fieldvalue}
+  ${reply}=  Call Method  ${USERS.users['${username}'].client}  patch_contract  ${tender}  ${contract}
+  Log  ${reply}
+
+
+Вказати дату початку дії угоди
+  [Arguments]  ${username}  ${tender_uaid}  ${contract_index}  ${fieldvalue}
+  ${tender}=  openprocurement_client.Пошук тендера по ідентифікатору  ${username}  ${tender_uaid}
+  ${period}=  Create Dictionary  startDate=${fieldvalue}
+  ${contract}=  Create Dictionary  data=${tender.data.contracts[${contract_index}]}
```
Comment (kosaniak) [inline op_robot_tests/tests_files/brokers/openprocurement_client.robot]: якщо в попередньому ключовому слові був створений ${contract.data} , то тут він вже недоступний?

File: op_robot_tests/tests_files/brokers/openprocurement_client.robot [hunk 7]
```diff
@@ -1069,3 +1056,71 @@ Library  openprocurement_client_helper.py
...
+Вказати дату завершення дії угоди
+  [Arguments]  ${username}  ${tender_uaid}  ${contract_index}  ${fieldvalue}
+  ${tender}=  openprocurement_client.Пошук тендера по ідентифікатору  ${username}  ${tender_uaid}
+  ${period}=  Create Dictionary  endDate=${fieldvalue}
+  ${contract}=  Create Dictionary  data=${tender.data.contracts[${contract_index}]}
+  Set To Dictionary  ${contract.data}  period=${period}
+  Log  ${contract}
+  ${reply}=  Call Method  ${USERS.users['${username}'].client}  patch_contract  ${tender}  ${contract}
+  Log  ${reply}
+
```
Comment (kosaniak) [inline op_robot_tests/tests_files/brokers/openprocurement_client.robot]: не дуже мені подобається, що *Вказати дату початку дії угоди* і *Вказати дату завершення дії угоди* Практично однакові. Можливо об'єднаємо їх

Comment (OrysiaDrabych) [inline op_robot_tests/tests_files/brokers/openprocurement_client.robot]: Об'єднала

**Comment by OrysiaDrabych** [comment]
DateSigned and Period fields to be removed from cache later if needed

**Pull request merged**